OrbitQt/OrbitStartupWindow.h

```diff
@@ -0,0 +1,29 @@
+#ifndef ORBITQT_ORBIT_STARTUP_WINDOW_H_
+#define ORBITQT_ORBIT_STARTUP_WINDOW_H_
+
+#include <QDialog>
+#include <QPointer>
+#include <QWidget>
+#include <optional>
+
+#include "GgpClient.h"
+#include "GgpInstance.h"
+#include "GgpInstanceItemModel.h"
+
+class OrbitStartupWindow : public QDialog {
+
+ public:
+  explicit OrbitStartupWindow(QWidget* parent = nullptr);
+  int Run(std::string* ip_address);
+
+ private:
+  void ReloadInstances();
+
+  std::optional<GgpClient> ggp_client_;
+  std::optional<GgpInstance> chosen_instance_;
+  QPointer<GgpInstanceItemModel> model_;
+
+  GgpInstance localhost_instance_;
```

Comment:
I don't get this. How can a GgpInstance be a localhost instance? We don't run the UI on a gamelet, right?

Reply:
This is to have the local machine as an option to choose for profiling. 

I admit this is a rather hotfix solution. Maybe a base class called something like "Profiling Target" and subclass the GgpInstance from it is the better solution. I could do that, or leave it as is. Any thoughts?

Reply:
Ahhh, got it. At least I would change the name in this case.

Personally I would not put static data into a variable which is not supposed to be changed.
I would add the entry directly to the Model or even better add a proxy model. I also don't see why
this needs to be a member variable. You're not passing a pointer to this somewhere and have to preserve lifetime, right?

Reply:
I agree with you that this is not a good solution. I am not familiar with proxy models, but I am eager to learn what you think the best way of handling this is. Sadly I did not find useful things via google. Would you mind pointing me somewhere or explain what you would do?

Reply:
For proxy models you can have a look here: https://doc.qt.io/qt-5/qabstractproxymodel.html
https://doc.qt.io/qt-5/model-view-programming.html

But be aware that this is quite some work. I would suggest to go for one of the simpler fixes for now.

Reply:
ptal, its now a const variable that lives only in OrbitStartupWindow.cpp. I am unsure if how I did it is a good idea. Please let me know if there is a better (and quick) option.